pkg/gofr/cmd/terminal/progress.go

```diff
@@ -0,0 +1,107 @@
+package terminal
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+type ProgressBar struct {
+	stream  Out
+	current int64
+	total   int64
+	tWidth  int
+	mu      sync.Mutex
+}
+
+type Term interface {
+	IsTerminal(fd int) bool
+	GetSize(fd int) (width, height int, err error)
+}
+
+func NewProgressBar(out Out, total int64) *ProgressBar {
+	w, _, err := out.getSize()
+	if err != nil {
+		fmt.Printf("error getting terminal size, err : %v, could not initialize progress bar\n", err)
+	}
+
+	if total < 0 {
+		fmt.Println("error initializing progress bar, total should be > 0")
+
+		total = 0
+	}
+
+	return &ProgressBar{
+		stream:  out,
+		total:   total,
+		tWidth:  w,
+		current: 0,
+	}
+}
+
+func (p *ProgressBar) Incr(i int64) bool {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	if p.current < p.total {
+		p.current += i
+		if p.current > p.total {
+			p.current = p.total
+		}
+
+		p.updateProgressBar()
+	}
+
+	return p.current != p.total
+}
+
+func (p *ProgressBar) updateProgressBar() {
+	// perform the TUI update of the progress bar
+	p.stream.Print("\r")
+
+	pString := p.getString()
+	p.stream.Printf("%s", pString)
```

Comment:
This is a print then, no?


```suggestion
	p.stream.Print(pString)
```